Readme.md

```diff
@@ -168,6 +171,119 @@ This one needs special mention due to how useful it is. It basically is a window
 Yes, vysor was great, but if you want to share your keyboard and mouse directly to your Android device, then this app is for you. It enables you to control your Android device as if it was part of your desktop computer. The [FREE version](https://play.google.com/store/apps/details?id=com.floriandraschbacher.deskdock.free) includes the use of computer mouse, while the [PRO version](https://play.google.com/store/apps/details?id=com.floriandraschbacher.deskdock.pro) includes features such as use of keyboard. This is useful where you can test your app without your hands ever leaving your keyboard.
 
 [<p align="right">Back to Index</p>](#index)
+
+#### ***ADB SHELL***
+Few handy commands you can use to interact with emulator/device, through terminal.
+	
++  List all connected devices
+
+  ```sh
+	adb devices
+  
+  ```
+  
++ Install an application on device 
+	- -s to install on the SD Card
+	- -g to grant all permissions listed in the manifest [[More Info]](https://developer.android.com/studio/command-line/adb.html)
+
+  ```sh
+  adb install -r ~/application.apk
+  
+  ```
+  
++ Uninstall an application
+
+  ```sh
+  adb uninstall your.app.package
+  
+  ```
+  
++ Download a file from device
+
+  ```sh
+  	adb pull /sdcard/file.ext
+  
+  ```
+  
++ Upload a file to a device
+
+  ```sh
+	adb push file.ext /sdcard
+  
+  ```
+  
++ Print all installed packages on device
+
+  ```sh
+	adb shell pm list packages -f
+  
+  ```
+  
++ Clear Application data
+
+  ```sh
+  adb shell pm clear your.app.package
+  
+  ```
+  
++ Toggle Data and Wifi for device.
+	
+	```sh
+	adb shell svc data disable
+	  
+	```
+	
+	```sh
+	adb shell svc wifi enable
+	  
+	```
+  
++ Location of Device
+	- High accuracy: gps,wifi,network or gps,network (in case you want the wifi)
+	- Battery saving: wifi,network or network only
+	- Device only: gps 
+	
+	```sh
+	adb shell settings put secure location_providers_allowed gps
+	  
+	```
++ Reset all granted permissions.
+
+	```sh
+	adb shell pm reset-permissions -p your.app.package
+	  
+	```
++ Reset a specific permission
+  
+	```
+	adb shell pm revoke your.app.package android.permission.WRITE_EXTERNAL_STORAGE
+	  
+	```
++ Broadcast Actions
+  	  
+	```sh
+	adb shell am broadcast -a 'my_action' 
+	  
+	```
++ [Simulating Android killing your app in the background] (https://twitter.com/Jahnold/status/759775495655333888)
```

Comment:
There is an extra space between `]` and `(` , which makes the link not render properly. The extra space should be removed.